[chore] add shared examples for files info query tests

Kharonus opened the pull request:
# What are you trying to accomplish?
- use shared examples to make test setups of query tests similar

Kharonus committed:
[chore] add shared examples for files info query tests

Kharonus commented:
⚠️ Here surfaced an old issue again: NC query result is quite different to the result from OD and SP. Eventually we should solve that. Findings:

- `permission` is empty in OD and SP, and a mixture of letters in NC
- `status` is a symbol on 4xx responses in OD/SP, but a string elsewhere
- `status` is written lowercase in OD/SP, and as uppercase words in NC

Should we maybe create a code maintenance for it?

NobodysNightmare commented on modules/storages/spec/support/shared_examples_for_storage_providers/files_info_query_examples.rb:
```diff
@@ -0,0 +1,59 @@
+
+    response = result.value!
+    expect(response).to be_an(Array)
+    response.each { expect(it).to be_a(Storages::Adapters::Results::StorageFileInfo) }
```
:yellow_circle: I think this is supported by RSpec as

```suggestion
    expect(response).to all(be_a(Storages::Adapters::Results::StorageFileInfo))
```

NobodysNightmare commented on modules/storages/spec/support/shared_examples_for_storage_providers/files_info_query_examples.rb:
```diff
@@ -0,0 +1,59 @@
+    response = result.value!
+    expect(response).to be_an(Array)
+    response.each { expect(it).to be_a(Storages::Adapters::Results::StorageFileInfo) }
+    expect(response.size).to eq(file_infos.size)
```
:green_circle: Isn't this expectation kinda duplicated to the one below (but it will fail the spec earlier with a less specific message)?

Below we expect equality between two arrays. This includes order and amount of items. But seeing

```
expected: [1, 2, 3]
got: [1, 2]
```

tells me much more, because i know which element is missing.

Kharonus replied:
err, definitely. don't know, what I was thinking back then. :D

NobodysNightmare commented on modules/storages/spec/support/shared_examples_for_storage_providers/files_info_query_examples.rb:
```diff
@@ -0,0 +1,59 @@
+    expect(response).to be_an(Array)
+    response.each { expect(it).to be_a(Storages::Adapters::Results::StorageFileInfo) }
+    expect(response.size).to eq(file_infos.size)
+    expect(response).to eq(file_infos)
```
:yellow_circle: I'd prefer if we called the `file_infos` something like `expected_file_infos` or `expected_result`.

My focus is on the prefix of `expected_`  here, because from the outside of this shared example (i.e. from the spec using it), it's pretty unclear whether `file_infos` is some input to the shared example. If it was called `expected_something`, I'd immediately get the idea, that it's not an input, but rather the expected output.

Kharonus replied:
agreed. and changed.

NobodysNightmare commented on modules/storages/spec/common/storages/adapters/providers/nextcloud/queries/files_info_query_spec.rb:
```diff
@@ -40,72 +40,81 @@ module Queries
-                    expect(result.value!).to eq([])
-                  end
-                end
+            context "with outbound request successful", vcr: "nextcloud/files_info_query_success" do
```
:yellow_circle: Is this the correct description for the context? I didn't look into the VCR cassette, but I am surprising that the request is successful when two out of three file infos come back with an HTTP 403.

NobodysNightmare reviewed (approved):
Overall I like the improvement made through the shared examples.

Some context descriptions and variables names could be more clear, I noted that down.

Kharonus committed:
[chore] improved test context

- better context descriptions for nextcloud tests
- better rspec expectations
- better variable naming
